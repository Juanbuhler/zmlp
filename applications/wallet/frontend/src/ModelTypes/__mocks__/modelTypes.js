const modelTypes = {
  results: [
    {
      name: 'ZVI_KNN_CLASSIFIER',
      label: 'Sci-kit Learn KNN Classifier',
      description:
        'Classify images or documents using a KNN classifier.  This type of model generates a single prediction which can be used to quickly organize assets into general groups.The KNN classifier works with just a single image and label.',
      objective: 'Label Detection',
      provider: 'Zorroa',
      deployOnTrainingSet: true,
      minConcepts: 0,
      minExamples: 0,
    },
    {
      name: 'ZVI_LABEL_DETECTION',
      label: 'Tensorflow CNN Classifier',
      description:
        'Classify images or documents using a custom strained CNN deep learning algorithm.  This type of modelgenerates multiple predictions and can be trained to identify very specific features. The label detection classifier requires at least 2 concepts with 10 labeled images each.',
      objective: 'Label Detection',
      provider: 'Zorroa',
      deployOnTrainingSet: false,
      minConcepts: 2,
      minExamples: 10,
    },
    {
      name: 'TF2_IMAGE_CLASSIFIER',
      label: 'Tensorflow2 Keras Image Classifier',
      description: 'Upload a Tensorflow model to use for image classification.',
      objective: 'Label Detection',
      provider: 'Google',
      deployOnTrainingSet: true,
      minConcepts: 0,
      minExamples: 0,
    },
    {
      name: 'GCP_LABEL_DETECTION',
      label: 'Google AutoML Classifier',
      description: 'Utilize Google AutoML to train an image classifier.',
      objective: 'Label Detection',
      provider: 'Google',
      deployOnTrainingSet: true,
      minConcepts: 2,
      minExamples: 10,
    },
  ],
}

export default modelTypes
