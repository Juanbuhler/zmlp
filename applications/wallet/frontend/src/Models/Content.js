import PropTypes from 'prop-types'
import useSWR from 'swr'

import ModelsTable from './Table'

const ModelsContent = ({ projectId }) => {
  const {
    data: { results: modelTypes },
  } = useSWR(`/api/v1/projects/${projectId}/models/model_types/`)

  return <ModelsTable projectId={projectId} modelTypes={modelTypes} />
}

ModelsContent.propTypes = {
  projectId: PropTypes.string.isRequired,
}

export default ModelsContent
