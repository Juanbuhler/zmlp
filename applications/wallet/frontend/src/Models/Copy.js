import PropTypes from 'prop-types'
import Link from 'next/link'

import { constants, colors } from '../Styles'

const ModelsCopy = ({ projectId }) => {
  return (
    <p
      css={{
        maxWidth: constants.paragraph.maxWidth,
        color: colors.structure.zinc,
      }}
    >
      FINAL COPY NEEDED Steps in adding labels and then coming back to the model
      detail page to train. Once a model has been trained it will be available
      to add to{' '}
      <Link
        href="/[projectId]/data-sources"
        as={`/${projectId}/data-sources`}
        passHref
      >
        <a css={{ color: colors.key.one }}>data sources</a>
      </Link>
      . Learn more about{' '}
      <a
        css={{ color: colors.key.one }}
        target="_blank"
        rel="noopener noreferrer"
        href="https://app.gitbook.com/@zorroa/s/zmlp/client/data-sets"
      >
        custom models
      </a>
      .
    </p>
  )
}

ModelsCopy.propTypes = {
  projectId: PropTypes.string.isRequired,
}

export default ModelsCopy
