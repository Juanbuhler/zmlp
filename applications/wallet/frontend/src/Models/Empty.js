import { typography } from '../Styles'

import NoJobsSvg from '../Icons/noJobs.svg'

const ModelsEmpty = () => {
  return (
    <div
      css={{
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        fontSize: typography.size.large,
        lineHeight: typography.height.large,
      }}
    >
      <NoJobsSvg width={200} />
      <div>There are currently no custom models.</div>
    </div>
  )
}

export default ModelsEmpty
