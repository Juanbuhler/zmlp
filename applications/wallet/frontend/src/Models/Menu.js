import { useState } from 'react'
import PropTypes from 'prop-types'

import Menu from '../Menu'
import Button, { VARIANTS } from '../Button'
import ButtonActions from '../Button/Actions'
import ModelDeleteModal from '../Model/DeleteModal'

const ModelsMenu = ({ projectId, modelId, name }) => {
  const [isDeleteModalOpen, setDeleteModalOpen] = useState(false)

  return (
    <>
      <Menu open="bottom-left" button={ButtonActions}>
        {({ onBlur, onClick }) => (
          <div>
            <ul>
              <li>
                <Button
                  variant={VARIANTS.MENU_ITEM}
                  onBlur={onBlur}
                  onClick={() => {
                    onClick()
                    setDeleteModalOpen(true)
                  }}
                >
                  Delete
                </Button>
              </li>
            </ul>
          </div>
        )}
      </Menu>

      <ModelDeleteModal
        projectId={projectId}
        modelId={modelId}
        name={name}
        isDeleteModalOpen={isDeleteModalOpen}
        setDeleteModalOpen={setDeleteModalOpen}
      />
    </>
  )
}

ModelsMenu.propTypes = {
  projectId: PropTypes.string.isRequired,
  modelId: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
}

export default ModelsMenu
