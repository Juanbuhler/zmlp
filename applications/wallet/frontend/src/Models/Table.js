import PropTypes from 'prop-types'

import Table, { ROLES } from '../Table'

import ModelsEmpty from './Empty'
import ModelsRow from './Row'

const ModelsTable = ({ projectId, modelTypes }) => {
  return (
    <Table
      role={ROLES.ML_Tools}
      legend="Models"
      url={`/api/v1/projects/${projectId}/models/`}
      refreshKeys={[]}
      refreshButton={false}
      columns={['Name', 'Type', 'Module']}
      expandColumn={0}
      renderEmpty={<ModelsEmpty />}
      renderRow={({ result }) => (
        <ModelsRow
          key={result.id}
          projectId={projectId}
          model={result}
          modelTypes={modelTypes}
        />
      )}
    />
  )
}

ModelsTable.propTypes = {
  projectId: PropTypes.string.isRequired,
  modelTypes: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      label: PropTypes.string.isRequired,
    }).isRequired,
  ).isRequired,
}

export default ModelsTable
