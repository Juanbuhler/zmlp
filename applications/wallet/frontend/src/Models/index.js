/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
import Head from 'next/head'
import { useRouter } from 'next/router'
import Link from 'next/link'

import { spacing } from '../Styles'

import { useLocalStorage } from '../LocalStorage/helpers'
import { SCOPE_OPTIONS } from '../AssetLabeling/helpers'

import PageTitle from '../PageTitle'
import BetaBadge from '../BetaBadge'
import FlashMessage, { VARIANTS as FLASH_VARIANTS } from '../FlashMessage'
import Tabs from '../Tabs'
import SuspenseBoundary, { ROLES } from '../SuspenseBoundary'

import ModelsContent from './Content'

const Models = () => {
  const {
    query: { projectId, action, modelId },
  } = useRouter()

  const [, setPanel] = useLocalStorage({
    key: 'leftOpeningPanel',
  })

  const [, setModelFields] = useLocalStorage({
    key: `AssetLabelingAdd.${projectId}`,
    reducer: (state, a) => ({ ...state, ...a }),
    initialState: {
      modelId: modelId || '',
      label: '',
      scope: '',
    },
  })

  return (
    <>
      <Head>
        <title>Custom Models</title>
      </Head>

      <PageTitle>
        <BetaBadge isLeft />
        Custom Models
      </PageTitle>

      {action === 'add-model-success' && (
        <div css={{ display: 'flex', paddingTop: spacing.base }}>
          <FlashMessage variant={FLASH_VARIANTS.SUCCESS}>
            Model created.{' '}
            <Link
              href="/[projectId]/visualizer"
              as={`/${projectId}/visualizer`}
              passHref
            >
              <a
                onClick={() => {
                  setPanel({ value: 'assetLabeling' })

                  setModelFields({
                    modelId,
                    scope: SCOPE_OPTIONS[0].value,
                    label: '',
                  })
                }}
              >
                Start Labeling
              </a>
            </Link>
          </FlashMessage>
        </div>
      )}

      {action === 'delete-model-success' && (
        <div css={{ display: 'flex', paddingTop: spacing.base }}>
          <FlashMessage variant={FLASH_VARIANTS.SUCCESS}>
            Model deleted.
          </FlashMessage>
        </div>
      )}

      <Tabs
        tabs={[
          { title: 'View all', href: '/[projectId]/models' },
          { title: 'Create New Model', href: '/[projectId]/models/add' },
        ]}
      />

      <SuspenseBoundary role={ROLES.ML_Tools}>
        <ModelsContent projectId={projectId} />
      </SuspenseBoundary>
    </>
  )
}

export default Models
