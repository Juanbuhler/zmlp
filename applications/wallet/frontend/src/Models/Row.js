import PropTypes from 'prop-types'
import Link from 'next/link'

import { onRowClickRouterPush } from '../Table/helpers'

import ModelsMenu from './Menu'

const ModelsRow = ({
  projectId,
  model: { id: modelId, name, type, moduleName },
  modelTypes,
}) => {
  const { label } = modelTypes.find(({ name: n }) => n === type) || {
    label: type,
  }

  return (
    <tr
      css={{ cursor: 'pointer' }}
      onClick={onRowClickRouterPush(
        '/[projectId]/models/[modelId]',
        `/${projectId}/models/${modelId}`,
      )}
    >
      <td>
        <Link
          href="/[projectId]/models/[modelId]"
          as={`/${projectId}/models/${modelId}`}
          passHref
        >
          <a css={{ ':hover': { textDecoration: 'none' } }}>{name}</a>
        </Link>
      </td>

      <td>{label}</td>

      <td>{moduleName}</td>

      <td>
        <ModelsMenu projectId={projectId} modelId={modelId} />
      </td>
    </tr>
  )
}

ModelsRow.propTypes = {
  projectId: PropTypes.string.isRequired,
  model: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    moduleName: PropTypes.string.isRequired,
  }).isRequired,
  modelTypes: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      label: PropTypes.string.isRequired,
    }).isRequired,
  ).isRequired,
}

export default ModelsRow
