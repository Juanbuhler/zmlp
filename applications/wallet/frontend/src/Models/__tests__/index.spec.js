import TestRenderer from 'react-test-renderer'

import mockUser from '../../User/__mocks__/user'
import modelTypes from '../../ModelTypes/__mocks__/modelTypes'

import User from '../../User'

import Models from '..'

const PROJECT_ID = '76917058-b147-4556-987a-0a0f11e46d9b'

jest.mock('next/link', () => 'Link')
jest.mock('../Table', () => 'ModelsTable')

describe('<Models />', () => {
  it('should render properly', async () => {
    require('next/router').__setUseRouter({
      pathname: `/[projectId]/models`,
      query: {
        projectId: PROJECT_ID,
        action: 'delete-model-success',
      },
    })

    require('swr').__setMockUseSWRResponse({ data: modelTypes })

    const component = TestRenderer.create(
      <User initialUser={mockUser}>
        <Models />
      </User>,
    )

    expect(component.toJSON()).toMatchSnapshot()
  })
})
