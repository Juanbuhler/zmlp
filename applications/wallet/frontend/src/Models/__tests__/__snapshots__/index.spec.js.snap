// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Models /> should let the user know a model was deleted 1`] = `
Array [
  .emotion-0 {
  font-size: 18px;
  line-height: 21px;
  font-weight: 400;
  padding-top: 24px;
  padding-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  color: #ffffff;
  background-color: #d6680b;
  border-radius: 32px;
  font-size: 11px;
  line-height: 15px;
  font-weight: 500;
  text-transform: uppercase;
  letter-spacing: 0.2px;
  padding: 1px;
  padding-left: 6px;
  padding-right: 6px;
  margin-right: 8px;
}

<h2
    className="emotion-0"
  >
    <span
      className="emotion-1"
    >
      beta
    </span>
    Custom Models
  </h2>,
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 8px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  background-color: #e3f7e8;
  border-radius: 2px;
  padding: 12px;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding-left: 12px;
  color: #191919;
  font-weight: 500;
  white-space: pre-line;
}

.emotion-2 a {
  color: #191919;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  margin-left: 4px;
}

.emotion-2 a:hover {
  color: #3A5DFF;
}

<div
    className="emotion-0"
  >
    <div
      className="emotion-1"
    >
      <svg
        color="#009f22"
        data-file-name="checkmark.svg"
        height={20}
      />
      <div
        className="emotion-2"
        role="alert"
      >
        Model deleted.
      </div>
    </div>
  </div>,
  .emotion-0 {
  padding: 16px;
  padding-left: 0;
  padding-right: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-1 {
  list-style-type: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-bottom: 1px solid #4a4a4a;
}

.emotion-2 {
  padding-right: 16px;
  margin-bottom: -1px;
}

.emotion-3 {
  border: 0 #3A5DFF solid;
  border-bottom-width: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 8px;
  padding-bottom: 8px;
  text-transform: uppercase;
  font-size: 17px;
  line-height: 20px;
  font-weight: 700;
}

.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-5 {
  border: 0 #3A5DFF solid;
  border-bottom-width: 0;
  color: #b3b3b3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 8px;
  padding-bottom: 8px;
  text-transform: uppercase;
  font-size: 17px;
  line-height: 20px;
  font-weight: 700;
}

.emotion-5:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #3A5DFF;
}

<nav
    className="emotion-0"
  >
    <ul
      className="emotion-1"
    >
      <li
        className="emotion-2"
      >
        <Link
          as="/76917058-b147-4556-987a-0a0f11e46d9b/models"
          href="/[projectId]/models"
          passHref={true}
        >
          <a
            className="emotion-3"
            onClick={[Function]}
          >
            View all
          </a>
        </Link>
      </li>
      <li
        className="emotion-2"
      >
        <Link
          as="/76917058-b147-4556-987a-0a0f11e46d9b/models/add"
          href="/[projectId]/models/add"
          passHref={true}
        >
          <a
            className="emotion-5"
            onClick={[Function]}
          >
            Create New Model
          </a>
        </Link>
      </li>
    </ul>
  </nav>,
  <ModelsTable
    modelTypes={
      Array [
        Object {
          "deployOnTrainingSet": true,
          "description": "Classify images or documents using a KNN classifier.  This type of model generates a single prediction which can be used to quickly organize assets into general groups.The KNN classifier works with just a single image and label.",
          "label": "Sci-kit Learn KNN Classifier",
          "minConcepts": 0,
          "minExamples": 0,
          "name": "ZVI_KNN_CLASSIFIER",
          "objective": "Label Detection",
          "provider": "Zorroa",
        },
        Object {
          "deployOnTrainingSet": false,
          "description": "Classify images or documents using a custom strained CNN deep learning algorithm.  This type of modelgenerates multiple predictions and can be trained to identify very specific features. The label detection classifier requires at least 2 concepts with 10 labeled images each.",
          "label": "Tensorflow CNN Classifier",
          "minConcepts": 2,
          "minExamples": 10,
          "name": "ZVI_LABEL_DETECTION",
          "objective": "Label Detection",
          "provider": "Zorroa",
        },
        Object {
          "deployOnTrainingSet": true,
          "description": "Upload a Tensorflow model to use for image classification.",
          "label": "Tensorflow2 Keras Image Classifier",
          "minConcepts": 0,
          "minExamples": 0,
          "name": "TF2_IMAGE_CLASSIFIER",
          "objective": "Label Detection",
          "provider": "Google",
        },
        Object {
          "deployOnTrainingSet": true,
          "description": "Utilize Google AutoML to train an image classifier.",
          "label": "Google AutoML Classifier",
          "minConcepts": 2,
          "minExamples": 10,
          "name": "GCP_LABEL_DETECTION",
          "objective": "Label Detection",
          "provider": "Google",
        },
      ]
    }
    projectId="76917058-b147-4556-987a-0a0f11e46d9b"
  />,
]
`;

exports[`<Models /> should let the user start labeling 1`] = `
Array [
  .emotion-0 {
  font-size: 18px;
  line-height: 21px;
  font-weight: 400;
  padding-top: 24px;
  padding-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  color: #ffffff;
  background-color: #d6680b;
  border-radius: 32px;
  font-size: 11px;
  line-height: 15px;
  font-weight: 500;
  text-transform: uppercase;
  letter-spacing: 0.2px;
  padding: 1px;
  padding-left: 6px;
  padding-right: 6px;
  margin-right: 8px;
}

<h2
    className="emotion-0"
  >
    <span
      className="emotion-1"
    >
      beta
    </span>
    Custom Models
  </h2>,
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 8px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  background-color: #e3f7e8;
  border-radius: 2px;
  padding: 12px;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding-left: 12px;
  color: #191919;
  font-weight: 500;
  white-space: pre-line;
}

.emotion-2 a {
  color: #191919;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  margin-left: 4px;
}

.emotion-2 a:hover {
  color: #3A5DFF;
}

<div
    className="emotion-0"
  >
    <div
      className="emotion-1"
    >
      <svg
        color="#009f22"
        data-file-name="checkmark.svg"
        height={20}
      />
      <div
        className="emotion-2"
        role="alert"
      >
        Model created.
         
        <Link
          as="/76917058-b147-4556-987a-0a0f11e46d9b/visualizer"
          href="/[projectId]/visualizer"
          passHref={true}
        >
          <a
            onClick={[Function]}
          >
            Start Labeling
          </a>
        </Link>
      </div>
    </div>
  </div>,
  .emotion-0 {
  padding: 16px;
  padding-left: 0;
  padding-right: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-1 {
  list-style-type: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-bottom: 1px solid #4a4a4a;
}

.emotion-2 {
  padding-right: 16px;
  margin-bottom: -1px;
}

.emotion-3 {
  border: 0 #3A5DFF solid;
  border-bottom-width: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 8px;
  padding-bottom: 8px;
  text-transform: uppercase;
  font-size: 17px;
  line-height: 20px;
  font-weight: 700;
}

.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-5 {
  border: 0 #3A5DFF solid;
  border-bottom-width: 0;
  color: #b3b3b3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 8px;
  padding-bottom: 8px;
  text-transform: uppercase;
  font-size: 17px;
  line-height: 20px;
  font-weight: 700;
}

.emotion-5:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #3A5DFF;
}

<nav
    className="emotion-0"
  >
    <ul
      className="emotion-1"
    >
      <li
        className="emotion-2"
      >
        <Link
          as="/76917058-b147-4556-987a-0a0f11e46d9b/models"
          href="/[projectId]/models"
          passHref={true}
        >
          <a
            className="emotion-3"
            onClick={[Function]}
          >
            View all
          </a>
        </Link>
      </li>
      <li
        className="emotion-2"
      >
        <Link
          as="/76917058-b147-4556-987a-0a0f11e46d9b/models/add"
          href="/[projectId]/models/add"
          passHref={true}
        >
          <a
            className="emotion-5"
            onClick={[Function]}
          >
            Create New Model
          </a>
        </Link>
      </li>
    </ul>
  </nav>,
  <ModelsTable
    modelTypes={
      Array [
        Object {
          "deployOnTrainingSet": true,
          "description": "Classify images or documents using a KNN classifier.  This type of model generates a single prediction which can be used to quickly organize assets into general groups.The KNN classifier works with just a single image and label.",
          "label": "Sci-kit Learn KNN Classifier",
          "minConcepts": 0,
          "minExamples": 0,
          "name": "ZVI_KNN_CLASSIFIER",
          "objective": "Label Detection",
          "provider": "Zorroa",
        },
        Object {
          "deployOnTrainingSet": false,
          "description": "Classify images or documents using a custom strained CNN deep learning algorithm.  This type of modelgenerates multiple predictions and can be trained to identify very specific features. The label detection classifier requires at least 2 concepts with 10 labeled images each.",
          "label": "Tensorflow CNN Classifier",
          "minConcepts": 2,
          "minExamples": 10,
          "name": "ZVI_LABEL_DETECTION",
          "objective": "Label Detection",
          "provider": "Zorroa",
        },
        Object {
          "deployOnTrainingSet": true,
          "description": "Upload a Tensorflow model to use for image classification.",
          "label": "Tensorflow2 Keras Image Classifier",
          "minConcepts": 0,
          "minExamples": 0,
          "name": "TF2_IMAGE_CLASSIFIER",
          "objective": "Label Detection",
          "provider": "Google",
        },
        Object {
          "deployOnTrainingSet": true,
          "description": "Utilize Google AutoML to train an image classifier.",
          "label": "Google AutoML Classifier",
          "minConcepts": 2,
          "minExamples": 10,
          "name": "GCP_LABEL_DETECTION",
          "objective": "Label Detection",
          "provider": "Google",
        },
      ]
    }
    projectId="76917058-b147-4556-987a-0a0f11e46d9b"
  />,
]
`;
